stl/inc/ranges

```diff
@@ -3589,9 +3608,214 @@ namespace ranges {
     };
 
     template <class _Rng, class _Pat>
-    split_view(_Rng&&, _Pat&&) -> split_view<views::all_t<_Rng>, views::all_t<_Pat>>;
+    lazy_split_view(_Rng&&, _Pat&&) -> lazy_split_view<views::all_t<_Rng>, views::all_t<_Pat>>;
 
     template <input_range _Rng>
+    lazy_split_view(_Rng&&, range_value_t<_Rng>)
+        -> lazy_split_view<views::all_t<_Rng>, single_view<range_value_t<_Rng>>>;
+
+    namespace views {
+        // VARIABLE views::lazy_split
+        class _Lazy_split_fn {
+        private:
+            template <class _Delim>
+            struct _Partial : _Pipe::_Base<_Partial<_Delim>> {
+                /* [[no_unique_address]] */ _Delim _Delimiter;
+
+                template <viewable_range _Rng>
+                _NODISCARD constexpr auto operator()(_Rng&& _Range) const& noexcept(
+                    noexcept(lazy_split_view{_STD forward<_Rng>(_Range), _Delimiter})) requires requires {
+                    lazy_split_view{static_cast<_Rng&&>(_Range), _Delimiter};
+                }
+                { return lazy_split_view{_STD forward<_Rng>(_Range), _Delimiter}; }
+
+                template <viewable_range _Rng>
+                _NODISCARD constexpr auto operator()(_Rng&& _Range) && noexcept(noexcept(
+                    lazy_split_view{_STD forward<_Rng>(_Range), _STD forward<_Delim>(_Delimiter)})) requires requires {
+                    lazy_split_view{static_cast<_Rng&&>(_Range), static_cast<_Delim&&>(_Delimiter)};
+                }
+                { return lazy_split_view{_STD forward<_Rng>(_Range), _STD forward<_Delim>(_Delimiter)}; }
+            };
+
+        public:
+            // clang-format off
+            template <viewable_range _Rng, class _Pat>
+            _NODISCARD constexpr auto operator()(_Rng&& _Range, _Pat&& _Pattern) const noexcept(noexcept(
+                lazy_split_view{_STD forward<_Rng>(_Range), _STD forward<_Pat>(_Pattern)})) requires requires {
+                lazy_split_view{static_cast<_Rng&&>(_Range), static_cast<_Pat&&>(_Pattern)};
+            } {
+                // clang-format on
+                return lazy_split_view{_STD forward<_Rng>(_Range), _STD forward<_Pat>(_Pattern)};
+            }
+
+            // clang-format off
+            template <class _Delim>
+                requires is_lvalue_reference_v<_Delim> || move_constructible<_Delim>
+            _NODISCARD constexpr auto operator()(_Delim&& _Delimiter) const
+                noexcept(is_lvalue_reference_v<_Delim> || is_nothrow_move_constructible_v<_Delim>) {
+                // clang-format on
+                return _Partial<_Delim>{._Delimiter = _STD forward<_Delim>(_Delimiter)};
+            }

```

Comment:
This, and the `split` version in unchanged text below, are inconsistent with WG21-P2281. Some of the examples in the discussion section of that paper may be of interest.

Reply:
Ditto this will be fixed in a followup.